db/migrate/20181024163850_add_source_type.rb

```diff
@@ -0,0 +1,14 @@
+class AddSourceType < ActiveRecord::Migration[5.1]
+  def change
+    create_table :source_types, :id => :bigserial do |t|
+      t.string :name
+      t.string :display_name
+      t.string :vendor
+      t.jsonb :schema
+      t.timestamps
+      t.index :name, :unique => true
+    end
+
+    add_reference :sources, :source_type, :index => true, :null => false
```

Comment:
```add_foreign_key :sources, :source_type, on_delete: :cascade```

Not sure if we should do cascade delete, if somebody deletes a source type?

Reply:
Yeah that comes down to if we want to allow sources without an associated type.  I'm leaning towards not because then we open the door for third parties to be lazy and not explicitly define the source type.